Make it possible to turn off heaters

eirikwah opened the pull request:
I have implemented support for turning heaters off (and on again).

The rest of this text is from the updated `readme.md`:

---

There is no override in Nobø for "completely off", but there is a workaround.

Nobø heaters can normally not be set to override "off". This is not a limitation in pynobo, but a safety-mechanism in the Nobø system (maybe they don't want you to accidentally turn off all your heaters and get frozen pipes). However, it is possible to create a week profile that makes the heater "off" all the time. And then you can configure the system to switch a heater to this week profile to be able to turn the heater off.

If you tell hanobo the name of the "Off" week profile and the name for the normal ("On") week profile for your zones, you can use this module to turn off (and on) your heaters. The week profiles must already exist in your Nobø system. Use the Nobø app to create them and configure them correctly.

If you don't configure any `command_off` or `command_on` then turning off heaters will not be supported (and this may be fine for your use).

eirikwah committed:
Ignore JetBrains Rider project files

eirikwah committed:
Enable turning off heaters

echoromeo commented on README.md:
```diff
@@ -1,18 +1,31 @@
 The possible Operation modes are as following:
 * Auto - In this mode the Zone is in the Normal setting and Preset shows which state the Zone is in right now (according to calendar setup)
 * Heat - In this mode the Zone in in the Override setting and in the state selected by Preset (Away, Eco, Comfort)
+* Off - In this mode the Zone is in the Normal setting and will get a specially configured week program that turns off the heater (more info below) 
```
```suggestion
* Off - In this optional mode the Zone is in the Normal setting and will configure it to use a week profile that turns the heater(s) in the Zone completely off (more info below)
```

echoromeo commented on README.md:
```diff
@@ -1,18 +1,31 @@
 * Changing Preset to None will automatically change Operation to Auto and update Preset
 * Changing Operation to Auto will automatically update Preset
 * Changing Operation to Heat will set Preset to Comfort
 
```
```suggestion
* Changing Operation to Off will change Operation to Auto and change the week profile to the predefined "completely off" profile

```

echoromeo commented on README.md:
```diff
@@ -23,6 +36,13 @@ To get started with this superexperimental implementation:
         - platform: nobo_hub
           host: [your nobø serial] # You can use the 3 last digits if using discovery
       #    ip_address: [your nobø ip] # Uncomment if you do not want discovery
+          # command_off is the name of a week profile (that must exist in the Nobø system) that shall be used to turn off a heater in any zone:
+          command_off: Off all the time
+          # command_on is a dictionary (a list) on the form "zone name:week profile name" that shall be used to turn on a heater in different zones:
+          command_on:
+            Bedroom: Cold at night
+            Living room: Normal daily comfort
+
```
```suggestion
          # command_off: [your completely off week profile name] # Uncomment if you want to enable the completely off setting (bypassing the 7 degrees Away setting)
          # command_on: # Uncomment these if you want to enable the completely off setting, one line for each zone you want to allow bypassing the 7 degrees Away setting
          #   [zone name:return week profile name] 
          #   [zone name:return week profile name]
```

echoromeo commented on README.md:
```diff
@@ -1,18 +1,31 @@
 
+## There is no override in Nobø for "completely off", but there is a workaround
+
+Nobø heaters can normally not be set to override "off". This is not a limitation in [pynobo][pypi], but a safety-mechanism in the Nobø system (maybe they don't want you to accidentally turn off all your heaters and get frozen pipes). However, it is possible to create a week profile that makes the heater "off" all the time. And then you can configure the system to switch a heater to this week prof ...
+
+If you tell hanobo the name of the "Off" week profile and the name for the normal ("On") week profile for your zones, you can use this module to turn off (and on) your heaters. The week profiles must already exist in your Nobø system. Use the Nobø app to create them and configure them correctly.
```
```suggestion
Nobø heaters can normally not be set to override "off". This is not a limitation in [pynobo][pypi], but a safety-mechanism in the Nobø system (maybe they don't want you to accidentally turn off all your heaters and get frozen pipes). However, it is possible to create a week profile that makes the heaters "off" all the time. And then you can configure the system to switch a zone to this week profile to be able to turn the heater(s) off.

If you tell hanobo the name of the "Off" week profile and the name for the normal ("On") week profile for your zones, you can use this module to turn off (and on) your heaters. The week profiles must already exist in your Nobø system, and you need to list the "On" week profile for each zone in the nobo_hub configuration. Use the Nobø app to create them and configure them correctly.
```

echoromeo reviewed (changes requested):
Great work @eirikwah!
I have no use for this feature myself, but I do understand why some people will want to use it for certain zones.

I guess I only have review feedback on the readme, I hope my suggestions make sense?

Thanks, Egil

eirikwah committed:
Apply suggestions from code review (improve readme)

Improve the readability of the readme, and make it more in style with the existing text.

Co-authored-by: echoromeo <felis.lynx@gmail.com>

eirikwah commented:
> I have no use for this feature myself, but I do understand why some people will want to use it for certain zones.

Yes, this is not for everyone. :-) However, this will be very nice for anyone wanting to configure their own window sensor, so they can turn off the heater when the window is open (even when the outside temperature is below 7 degrees celsius). I will be using this for my "automatically close the window and then turn on the heater 30 minutes before my alarm clock goes off" project. 🤓 
 
> I guess I only have review feedback on the readme, I hope my suggestions make sense?

Yes! 👍 Thanks for taking your time to review this.
